pkg/distributor/distributor.go

```diff
@@ -178,10 +178,14 @@ func (cfg *Config) Validate(limits validation.Limits) error {
 		return errInvalidShardingStrategy
 	}
 
-	if cfg.ShardingStrategy == util.ShardingStrategyShuffle && limits.IngestionTenantShardSize <= 0 {
+	if limits.IngestionTenantShardSize < 0 {
```

Comment:
```
if cfg.ShardingStrategy == util.ShardingStrategyShuffle && limits.IngestionTenantShardSize < 0 {
}
```

I think it is fine to have `IngestionTenantShardSize < 0` when strategy is not shuffle sharding as it is a noop.